### ISSUE TITLE
incompatible type for argument 4 of 'cvRectangle'

### ISSUE THREAD

**Issue by sberryman** [AlexeyAB/yolo2_light#20]
yolo2_light: 5e5f9408cc9033e9043b25b00277924afc91c01f
OpenCV: 3.4.3
Cuda: 9.2
cudnn: 7

```shell
gcc  -DOPENCV `pkg-config --cflags opencv`  -DGPU -I/usr/local/cuda/include/ -DCUDNN  -Wall -Wfatal-errors -Ofast -DOPENCV -DGPU -DCUDNN -I/usr/local/cudnn/include -c ./src/yolov2_forward_network_quantized.c -o obj/yolov2_forward_network_quantized.o
./src/yolov2_forward_network_quantized.c: In function 'get_distribution':
./src/yolov2_forward_network_quantized.c:40:28: warning: unused variable 'max_val' [-Wunused-variable]
     float min_val = 10000, max_val = 0;
                            ^
./src/yolov2_forward_network_quantized.c:40:11: warning: unused variable 'min_val' [-Wunused-variable]
     float min_val = 10000, max_val = 0;
           ^
./src/yolov2_forward_network_quantized.c: In function 'draw_distribution':
./src/yolov2_forward_network_quantized.c:132:40: warning: implicit declaration of function 'CV_RGB' [-Wimplicit-function-declaration]
             cvRectangle(img, pt1, pt2, CV_RGB(128, 64, 32), CV_FILLED, 8, 0);
                                        ^
./src/yolov2_forward_network_quantized.c:132:40: error: incompatible type for argument 4 of 'cvRectangle'
compilation terminated due to -Wfatal-errors.
Makefile:103: recipe for target 'obj/yolov2_forward_network_quantized.o' failed
make: *** [obj/yolov2_forward_network_quantized.o] Error 1
```

I have to comment out quite a few lines from `src/yolov2_forward_network_quantized.c` to get it to compile. Basically everywhere CV_RGB is used. OpenCV appears to be dropping all C support, any thoughts about making this a C++ library instead? I've done a little wrapper and see you have added `extern "C" {` in a few of the headers already.

Edit: I'm aware you state `OpenCV <= 3.3.0` in the readme, commenting out all the lines with `CV_RGB` allows you to compile for 3.4.3.

**Comment by sberryman** [comment]
Easy for me to work around and assuming this isn't an issue for anyone else so I'll close it to clean up the list of outstanding issues.

**Comment by AlexeyAB** [comment]
@sberryman I added fix for compiling with OpenCV >= 3.4.0: https://github.com/AlexeyAB/yolo2_light/commit/1bc3ba78c45d0e7a17fa6f021f060bd4b7e23193#diff-4ddab5e6b8ac21eaed4f422abc8a0f09R94

Maybe it will work )
